### ISSUE TITLE
A way to use the 3dgs Viewer

### ISSUE THREAD

**Issue by m1ingshen** [hbb1/2d-gaussian-splatting#51]
I made some minor modifications to your code so that it can be used with the viewer provided by 3DGS. Here's my approach:
1. change the codes in function **construct_list_of_attributes** from **scene/gaussian_model.py** like this:
```
l = ['x', 'y', 'z', 'nx', 'ny', 'nz']
        # All channels except the 3 DC
        for i in range(self._features_dc.shape[1]*self._features_dc.shape[2]):
            l.append('f_dc_{}'.format(i))
        for i in range(self._features_rest.shape[1]*self._features_rest.shape[2]):
            l.append('f_rest_{}'.format(i))
        l.append('opacity')
        for i in range(self._scaling.shape[1] + 1):
            l.append('scale_{}'.format(i))
        ......
```
By this way,  the scales can have 3 dim, the same as 3dgs.
2. change the codes in fuction  **save_ply** from **scene/gaussian_model.py** like this:
```
mkdir_p(os.path.dirname(path))

        xyz = self._xyz.detach().cpu().numpy()
        normals = np.zeros_like(xyz)
        f_dc = self._features_dc.detach().transpose(1, 2).flatten(start_dim=1).contiguous().cpu().numpy()
        f_rest = self._features_rest.detach().transpose(1, 2).flatten(start_dim=1).contiguous().cpu().numpy()
        opacities = self._opacity.detach().cpu().numpy()
        scale = self._scaling.detach().cpu().numpy()
        rotation = self._rotation.detach().cpu().numpy()
        one_line = np.full((scale.shape[0],1), 0.0000000001)
        one_line = np.log(one_line)
        scale = np.concatenate((scale, one_line), axis=1)
        ......
```
By this way, the 2D gaussian is changed into 3D gaussian with a zeros_like dimension.

**Comment by hbb1** [comment]
Yes, have you observed severe perspective-distortion artifacts? Can you post some results here? By the way, we have made progress in creating a viewer.

**Comment by m1ingshen** [comment]
These are two examples of garden training 7_000 iterations
![image](https://github.com/hbb1/2d-gaussian-splatting/assets/59131946/ab962c63-5745-4ac5-bdae-2e8f0e4372e3)
![image](https://github.com/hbb1/2d-gaussian-splatting/assets/59131946/3f08406e-a62f-42cb-8953-250cb019ddf1)
I have only tested a few of models, maybe there are some distortions I haven't met.

**Comment by hbb1** [comment]
Thank you for reporting the results!  As I mentioned in the README, this is a possible but not perfect solution. Also, the additional scale will cause ambiguities.  Therefore, I think it is better to keep the current Gaussian Model. Let's keep the issue open for who would like a quick visualizer. Or let me refer people to your fork if you have.
